subprojects/ide-native/src/integTest/groovy/org/gradle/ide/visualstudio/fixtures/MSBuildExecutor.java

```diff
@@ -0,0 +1,104 @@
+
+import org.gradle.integtests.fixtures.executer.ExecutionFailure;
+import org.gradle.integtests.fixtures.executer.ExecutionResult;
+import org.gradle.integtests.fixtures.executer.OutputScrapingExecutionFailure;
+import org.gradle.integtests.fixtures.executer.OutputScrapingExecutionResult;
+import org.gradle.test.fixtures.file.ExecOutput;
+import org.gradle.test.fixtures.file.TestFile;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.gradle.ide.fixtures.IdeCommandLineUtil.buildEnvironment;
+import static org.testng.Assert.assertTrue;
+
+public class MSBuildExecutor {
+    public enum MSBuildAction {
+        BUILD,
+        CLEAN;
+
+        @Override
+        public String toString() {
+            return this.name().toLowerCase();
+        }
+    }
+
+    private final List<String> args = new ArrayList<String>();
+    private final TestFile testDirectory;
+
+    public MSBuildExecutor(TestFile testDirectory) {
+        this.testDirectory = testDirectory;
+    }
+
+    public MSBuildExecutor withSolution(SolutionFile visualStudioSolution) {
+        TestFile solutionFile = new TestFile(visualStudioSolution.getFile());
+        solutionFile.assertIsFile();
+        return addArguments(solutionFile.getAbsolutePath());
+    }
+
+    public MSBuildExecutor withConfiguration(String configurationName) {
+        return addArguments("/p:Configuration=" + configurationName);
+    }
+
+    public MSBuildExecutor withArgument(String arg) {
+        this.args.add(arg);
+        return this;
+    }
+
+    private MSBuildExecutor addArguments(String... args) {
+        this.args.addAll(Arrays.asList(args));
+        return this;
+    }
+
+    public ExecutionResult succeeds() {
+        return succeeds(MSBuildAction.BUILD);
+    }
+
+    public ExecutionResult succeeds(MSBuildAction action) {
+        withArgument("/t:" + action.toString());
+        ExecOutput result = findMSBuild().execute(args, buildEnvironment(testDirectory));
+        System.out.println(result.getOut());
+        return new OutputScrapingExecutionResult(result.getOut(), result.getError());
+    }
+
+    public ExecutionFailure fails() {
+        return fails(MSBuildAction.BUILD);
+    }
+
+    public ExecutionFailure fails(MSBuildAction action) {
+        withArgument(action.toString());
+        ExecOutput result = findMSBuild().execWithFailure(args, buildEnvironment(testDirectory));
+        System.out.println(result.getOut());
+        System.out.println(result.getError());
+        return new OutputScrapingExecutionFailure(result.getOut(), result.getError());
+    }
+
+    private TestFile findMSBuild() {
+        TestFile vswhere = new TestFile(System.getenv("ProgramFiles(x86)") + "/Microsoft Visual Studio/Installer/vswhere.exe");
+        assertTrue(vswhere.exists(), "This test requires vswhere to be installed in '%ProgramFiles(x86)%/Microsoft Visual Studio/Installer/vswhere.exe'");
```

Comment:
This is only true on x64 machines - on x32 machines, there won't be a `ProgramFiles(X86)` environment variable.  Can we use `CommandLineToolVersionLocator` for this (i.e. can we add something to `VisualStudioLocatorTestFixture` to make this vswhere easy to find?

Reply:
Sounds good, I will look into this.

Reply:
Done.